### PR TITLE
backport: bitcoin#18772, #20690, #20789, #20813, #21531, bitcoin-core/gui#13, #72, 115, #139, #171

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Regular backports from bitcoin v22

## What was done?
 - bitcoin-core/gui#13
 - bitcoin-core/gui#115
 - bitcoin/bitcoin#20690
 - bitcoin/bitcoin#18772
 - bitcoin/bitcoin#20813
 - bitcoin/bitcoin#20789
 - bitcoin-core/gui#171
 - bitcoin-core/gui#72
 - bitcoin-core/gui#139
 - bitcoin/bitcoin#21531

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by knst** [comment]
> check https://gitlab.com/dashpay/dash/-/jobs/6565831300

seems as random failure, because both times failed on different tests. Let's wait CI to run for rebased version

>  @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/23becc3efa5ab684d711cfd50e150ab6341629cd..e4583dee3b68088b82bf4057a9eaebefa209c64f) the bp-v22-p4 branch from[ 23becc3 ](https://github.com/dashpay/dash/commit/23becc3efa5ab684d711cfd50e150ab6341629cd)to e4583de

rebased on top of develop after master merged back

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
one tiny issue, LGTM otherwise

File: test/functional/example_test.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
-#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (knst) [reply test/functional/example_test.py:1]: good catch! I hope you did check all 300 files manually :) 
Fixed

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK a2f190deff19dc21ffff700ef3a13d5f53a92ae3

**Pull request merged**